ribbon-loadbalancer/src/main/java/com/netflix/client/ClientFactory.java

```diff
@@ -93,7 +93,14 @@
      * @throws RuntimeException if an error occurs in creating the client.
      */
     public static synchronized IClient getNamedClient(String name) {
-        return getNamedClient(name, DefaultClientConfigImpl.class);
+        if (simpleClientMap.get(name) != null) {
+            return simpleClientMap.get(name);
+        }
+        try {
+            return registerClientFromProperties(name, getNamedConfig(name));
```

Comment:
I see that `getNamedLoadBalancer` caches the one registered from config upon success, does this one intentionally not cache it in the map?

Reply:
It gets cached in registerClientFromProperties.  I really wanted to fix the cache interaction but the code is so convoluted that I'm afraid to break something.  I may revisit that at a later time.